packages/gatsby-plugin-canonical-urls/src/gatsby-browser.js

```diff
@@ -1,4 +1,10 @@
+const url = require(`url`)
```

Comment:
This seems heavy, is there a lighter way to parse the URL? Perhaps when server rendering we can just set an attribute with the protocol and host, so we don't have to parse the whole URL.